dagobah/core/delegator.py

```diff
@@ -0,0 +1,41 @@
+import logging
+
+from ..backend.base import BaseBackend
+
+logger = logging.getLogger('dagobah')
+
+class CommitDelegator(object):
+    """
+    The CommitDelegator is a class designed to handle any and all commits
+    to the database.
+
+    The delegator takes in commit requests from various objects, and decides
+    if they should actually be committed, and if so, manage any special
+    processing.
+    """
+
+    def __init__(self, backend=BaseBackend()):
+        self.backend = backend
+
+    def commit_dagobah(self, dagobah, cascade=False):
+        """ Commit this Dagobah instance to the backend.
+
+        If cascade is True, all child Jobs are commited as well.
+        """
+        logger.debug('Committing Dagobah instance with cascade={0}'.
+                     format(cascade))
+        self.backend.commit_dagobah(dagobah._serialize())
+        if cascade:
+            [self.commit_job(job) for job in dagobah.jobs]
```

Comment:
I'm curious as to why this is done the way it's done actually, you're right that it is extremely heavy on IO. I feel like an op like this should really be wrapped in a transaction, as committing a bunch of jobs and then dagobah could theoretically leave the DB in an invalid state if it breaks somewhere in between or something else crashes.

In the meantime, I'm going to try to disable committing dagobah in the cascade case and move the final dagobah commit to the end and see if it passes tests/looks okay when actually used